rest/route/host_spawn.go

```diff
@@ -789,6 +789,72 @@ func (h *deleteVolumeHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/volumes/{volume_id}
+
+type modifyVolumeHandler struct {
+	sc  data.Connector
+	env evergreen.Environment
+
+	volumeID string
+	opts     *host.VolumeModifyOptions
+}
+
+func makeModifyVolume(sc data.Connector) gimlet.RouteHandler {
+	return &modifyVolumeHandler{
+		sc: sc,
+	}
+}
+
+func (h *modifyVolumeHandler) Factory() gimlet.RouteHandler {
+	return &modifyVolumeHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *modifyVolumeHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+	h.opts = &host.VolumeModifyOptions{}
```

Comment:
You're reading json into a struct that doesn't have json tags. I would create a separate rest model for this if you need to use it in the db model, or just move it to the rest model if you don't

Reply:
I think that we shouldn't expose DB models as part of APIs ever.